segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageConfig.java

```diff
@@ -124,6 +135,32 @@
     @Getter
     final private int lateWarningThresholdInMillis;
 
+    /**
+     * Minimum delay in seconds between when garbage chunks are marked for deletion and actually deleted.
+     */
+    @Getter
+    final private Duration garbageCollectionDelay;
+
+    /**
+     * Number of chunks deleted concurrently.
+     * This number should be small enough so that it does interfere foreground requests.
```

Comment:
Rather than just advising, maybe you could a test to verify that this is not larger than half of the storage pool size or something along these lines. Then, you can decide whether to throw in this case (so, do not allow large thread pools for garbage collection) or log it as a warning.

Reply:
Storage GC is new . I don't know what is a good number here. We'll have to measure and get a sense for it.